src/main/java/org/sagebionetworks/bridge/services/PermissionService.java

```diff
@@ -0,0 +1,169 @@
+    private Set<EntityType> studyTypes = ImmutableSet.of(PARTICIPANTS, STUDY, STUDY_PI);
+    private Set<EntityType> assessmentTypes = ImmutableSet.of(ASSESSMENT);
+    
+    private PermissionDao permissionDao;
+    
+    private AccountService accountService;
+    
+    private OrganizationService organizationService;
+    
+    private StudyService studyService;
+    
+    private AssessmentService assessmentService;
+    
+    @Autowired
+    final void setPermissionDao(PermissionDao permissionDao) {
+        this.permissionDao = permissionDao;
+    }
+    
+    @Autowired
+    final void setAccountService(AccountService accountService) {
+        this.accountService = accountService;
+    }
+    
+    @Autowired
+    final void setOrganizationService(OrganizationService organizationService) {
+        this.organizationService = organizationService;
+    }
+    
+    @Autowired
+    final void setStudyService(StudyService studyService) {
+        this.studyService = studyService;
+    }
+    
+    @Autowired
+    final void setAssessmentService(AssessmentService assessmentService) {
+        this.assessmentService = assessmentService;
+    }
+    
+    // For mock testing
+    protected String generateGuid() {
+        return BridgeUtils.generateGuid();
+    }
+    
+    public PermissionDetail createPermission(String appId, Permission permission) {
+        checkArgument(isNotBlank(appId));
+        checkNotNull(permission);
+        
+        permission.setGuid(generateGuid());
+        
+        Validate.entityThrowingException(INSTANCE, permission);
+        
+        Permission createdPermission = permissionDao.createPermission(appId, permission);
+        
+        return getPermissionDetail(appId, createdPermission);
+    }
+    
+    public PermissionDetail updatePermission(String appId, Permission permission) {
+        checkArgument(isNotBlank(appId));
+        checkNotNull(permission);
+    
+        Validate.entityThrowingException(INSTANCE, permission);
+        
+        Permission existingPermission = permissionDao.getPermission(appId, permission.getGuid())
+                .orElseThrow(() -> new EntityNotFoundException(Permission.class));
+        
+        // Can only update accessLevel
+        permission.setAppId(existingPermission.getAppId());
+        permission.setUserId(existingPermission.getUserId());
+        permission.setEntityType(existingPermission.getEntityType());
+        permission.setEntityId(existingPermission.getEntityId());
+        
+        Permission updatedPermission = permissionDao.updatePermission(appId, permission);
+    
+        return getPermissionDetail(appId, updatedPermission);
+    }
+    
+    public List<PermissionDetail> getPermissionsForUser(String appId, String userId) {
+        checkArgument(isNotBlank(appId));
+        checkArgument(isNotBlank(userId));
+    
+        List<Permission> permissions = permissionDao.getPermissionsForUser(appId, userId);
```

Comment:
What happens if the user does not exist? (Should throw ENFE)

Reply:
Added a check for the user account before getting the permissions. Now throws ENFE.

Reply:
This still needs to be a ResourceList, not a list.